documentation/src/pages/recipes/data/recipes/full-stack-project-initializer.yaml

```diff
@@ -0,0 +1,242 @@
+version: 1.0.0
+title: Full-Stack Project Initializer
+description: An advanced recipe that orchestrates complete full-stack project initialization with intelligent framework detection, dependency management, code quality setup, and comprehensive documentation generation
+author:
+  contact: ARYPROGRAMMER
+
+activities:
+  - Analyze project requirements and detect optimal tech stack
+  - Initialize frontend and backend project structures
+  - Configure development environment with linting and testing
+  - Set up CI/CD pipeline and git repository
+  - Generate comprehensive documentation and README
+  - Install dependencies and verify project health
+
+instructions: |
+  You are a Full-Stack Project Initializer that creates production-ready project structures with best practices.
+  Create complete projects with frontend, backend, testing, linting, Docker, CI/CD, and documentation based on user parameters.
+
+parameters:
+  - key: project_name
+    input_type: string
+    requirement: required
+    description: Name of the project to initialize (alphanumeric and dashes only)
+  
+  - key: project_type
+    input_type: string
+    requirement: required
+    description: Type of project - options are 'fullstack', 'frontend', 'backend', 'cli', 'api'
+  
+  - key: frontend_framework
+    input_type: string
+    requirement: optional
+    default: "react"
+    description: Frontend framework to use (react, vue, svelte, nextjs, none)
+  
+  - key: backend_framework
+    input_type: string
+    requirement: optional
+    default: "nodejs"
+    description: Backend framework to use (nodejs, python-fastapi, go, rust-actix, none)
+  
+  - key: database
+    input_type: string
+    requirement: optional
+    default: "postgresql"
+    description: Database to integrate (postgresql, mongodb, mysql, sqlite, none)
+  
+  - key: include_docker
+    input_type: string
+    requirement: optional
+    default: "true"
+    description: Whether to include Docker configuration (true/false)
+  
+  - key: include_cicd
+    input_type: string
+    requirement: optional
+    default: "true"
+    description: Whether to include CI/CD pipeline (true/false)
+  
+  - key: include_testing
+    input_type: string
+    requirement: optional
+    default: "true"
+    description: Whether to include testing setup with unit and integration tests (true/false)
+  
+  - key: base_directory
+    input_type: string
+    requirement: optional
+    default: "."
+    description: Base directory where the project should be created
+
```

Comment:
do we need any awareness of what kind of system we're developing on? eg, mac vs windows may have different tooling or needs (for example, Windows will handle disk paths differently, one common area I had was building paths like c:\src\react-project where the `\r` in the path gets interpreted as a carriage return character, breaking other workflows.

maybe this doesn't need to be a parameter, it could just be added in the instructions that say something like "detect what kind of operating system I'm on, and if you detect that I'm on Windows, be careful about disk paths where backslash path separators aren't interpreted as escape characters in strings"